File: 2346-largest-3-same-digit-number-in-string/largest-3-same-digit-number-in-string.java
class Solution {
    public String largestGoodInteger(String num) {
        String result = "";
        
        for (int i = 0; i < num.length() - 2; i++)
            if (num.charAt(i) == num.charAt(i + 1) && num.charAt(i) == num.charAt(i + 2))
                if (num.substring(i, i + 3).compareTo(result) > 0)
                    result = num.substring(i, i + 3);

        return result;
    }
}

File: 2346-largest-3-same-digit-number-in-string/largest-3-same-digit-number-in-string.py
class Solution:
    def largestGoodInteger(self, num: str) -> str:
        result = ""

        for i in range(len(num) - 2):
            if num[i] == num[i + 1] and num[i] == num[i + 2]:
                if num[i : i + 3] > result:
                    result = num[i : i + 3]

        return result
